### PR TITLE
Add a download-artifact action

### PR DESCRIPTION
**Pull request by bjoernricks**
**What**:

Add a download-artifact action

**Why**:

In conjunction with the new trigger-workflow action this will replace the artifacts-download-trigger action.

**How**:

**Checklist**:

- [x] Tests
- [ ] CHANGELOG.md entry
- [x] Documentation

### CODE REVIEW

File: .github/workflows/test-download-artifacts-test-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create random file content
+        run: |
+          echo $RANDOM | sha256sum | cut -f 1 -d ' ' > content.txt
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: artifact
+          path: content.txt
```
Comment (y0urself) [inline .github/workflows/test-download-artifacts-test-workflow.yml:17]:
```suggestion
          path: content.txt
          retention-days: 2
```

File: download-artifact/action/artifact.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    FAILURE = "failure"
+    IN_PROGRESS = "in_progress"
+    NEUTRAL = "neutral"
+    QUEUED = "queued"
+    REQUESTED = "requested"
+    SKIPPED = "skipped"
+    STALE = "stale"
+    SUCCESS = "success"
+    TIMED_OUT = "timed_out"
+    WAITING = "waiting"
```
Comment (y0urself) [inline download-artifact/action/artifact.py:74]: Wouldn't it make sense to put this in `pontos.github.api.helper` and accessible over `pontos.github.api` in the `__init__.py` file?

**Review by y0urself — CHANGES_REQUESTED** [review]
Looks pretty solid! :)

File: download-artifact/action/artifact.py (hunk 2 above)
Comment (bjoernricks) [reply download-artifact/action/artifact.py:74]: Yes we should do that.

Comment (bjoernricks) [reply download-artifact/action/artifact.py:74]: https://github.com/greenbone/pontos/pull/431

**Review by y0urself — APPROVED** [review]

**Pull request merged**